src/app/providers/config/utils.ts

```diff
@@ -31,12 +32,12 @@ export const filterConfig = (
     } else {
       const config = addFederatedModulesToConfig(possibleDefaultConfigs[0], federatedModulesConfig);
       console.log('Done loading default config', config);
-      return config;
+      return [config, <boolean>possibleDefaultConfigs[0].fetchConfig];
```

Comment:
this cast is odd.

Reply:
fetchConfig can also be undefined, that's why, I could use something like `possibleDefaultConfigs[0].fetchConfig === undefined ? false : possibleDefaultConfigs[0].fetchConfig` but this is more compact in my opinion

Reply:
It's also semantically different. Casts are not the same as checks for undefineds.

Reply:
It might be better to introduce an object return here, rather than using an array?

Reply:
I opted for splitting the filterConfig and addFederateModules in this way:
https://github.com/redhat-developer/app-services-ui/pull/34/commits/0dece7df2199fa0befefa80623aac167195a034f
wdyt?